### PR TITLE
SRM remove multiple ENSGs per peptide

### PR DESCRIPTION
**Pull request by jaclynbeck-sage**
This PR handles the fact that multiple Ensembl IDs map to the same UniProt ID, so SRM data had duplicates with Ensembl IDs that were mostly HASGs. 

I ended up doing a bunch of cleanup across multiple notebooks as a side-effect and added it to this PR. So this PR contains the following changes:
1. **SRM processing now removes HASGs that match to peptide UniProt IDs**. I've confirmed that the only Ensembl IDs remaining for peptides that had duplicates are the ones in bold in [the JIRA ticket](https://sagebionetworks.jira.com/browse/AG-1296), and no other genes were removed from the data set. 
2. I made a new file `preprocessing_utils.py` and pulled some duplicated code into functions in that file. Right now, that's requests to Biomart and filtering HASGs, but can be expanded on if necessary. It currently lives in the same folder as the pre-processing notebooks.
3. Changed SRM pre-processing, GWAS pre-processing, and gene metadata pre-processing to use these functions instead of duplicating code. 
4. Gene metadata processing code was pretty out of date and needed some updating to work again. 
5. **Gene metadata processing now removes HASGs that are pulled in by Biomart ONLY** (it doesn't touch the ones that come from our data source files since we haven't decided how to handle that yet). The resulting `gene_table_merged.feather` file and `gene_info.json` (post-transformation) will have FEWER genes than the last update.  
6. Updating the `gene_table_merged.feather` file **pulled in Ensembl release 111**. 
7. The version numbers in both config files were changed to use the newly-generated SRM and gene table files. 

I spot-checked the new `gene_info.json` file generated by CI and it looks good to me (see also [this JIRA comment](https://sagebionetworks.jira.com/browse/AG-1296?focusedCommentId=195692) for why some HASGs are still in `gene_info`). 

The diffs on the notebooks are awful to look at so I've linked the actual notebook files below:
igap [before](https://github.com/Sage-Bionetworks/agora-data-tools/blob/6039932ec4a503fa176bfb6e366a5a72d8cfce6b/data_analysis/agora/notebooks/preprocessing/AG-796_process_gwas.ipynb) and [after](https://github.com/Sage-Bionetworks/agora-data-tools/blob/1bf18f28b805bdd889e0188e60a72600ebafaea3/data_analysis/agora/notebooks/preprocessing/AG-796_process_gwas.ipynb)
srm [before](https://github.com/Sage-Bionetworks/agora-data-tools/blob/620c094d4e73eb37f9797f329e8009173d401269/data_analysis/agora/notebooks/preprocessing/AG-866_process_srm_diffexp_data.ipynb) and [after](https://github.com/Sage-Bionetworks/agora-data-tools/blob/1bf18f28b805bdd889e0188e60a72600ebafaea3/data_analysis/agora/notebooks/preprocessing/AG-866_process_srm_diffexp_data.ipynb)
gene annotations [before](https://github.com/Sage-Bionetworks/agora-data-tools/blob/62fda2f739aec4c6aa1ac4c2ff638e3fa505cc39/data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb) and [after](https://github.com/Sage-Bionetworks/agora-data-tools/blob/1bf18f28b805bdd889e0188e60a72600ebafaea3/data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb)

### CODE REVIEW

File: data_analysis/agora/notebooks/preprocessing/AG-796_process_gwas.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-796_process_gwas.ipynb:325]: Here is where I moved biomart query code to preprocessing_utils.py

File: data_analysis/agora/notebooks/preprocessing/AG-796_process_gwas.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-796_process_gwas.ipynb:362]: Here is where I moved the HASG filter code to preprocessing_utils.py

File: data_analysis/agora/notebooks/preprocessing/AG-866_process_srm_diffexp_data.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-866_process_srm_diffexp_data.ipynb:45]: Most of the changes in this file are from the black formatter. I'll highlight actual code changes with comments.

File: data_analysis/agora/notebooks/preprocessing/AG-866_process_srm_diffexp_data.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-866_process_srm_diffexp_data.ipynb:485]: NEW code starts here to filter HASGs

File: data_analysis/agora/notebooks/preprocessing/AG-866_process_srm_diffexp_data.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-866_process_srm_diffexp_data.ipynb:860]: End new code

File: data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb:139]: This chunk is a code update

File: data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb:197]: End code update

File: data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb:357]: The couple changes above are a code update

File: data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb:389]: The couple lines above are a code update

File: data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb:897]: Code update above

File: data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb:945]: Code update above

File: data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb:1454]: Begin code update chunk

File: data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb:1494]: End code update chunk

File: data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb:2864]: Code update above

File: data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb
Comment (jaclynbeck-sage) [inline data_analysis/agora/notebooks/preprocessing/AG-896_Preprocess_Gene_Annotations.ipynb:2887]: Code update above

File: data_analysis/agora/notebooks/preprocessing/preprocessing_utils.py
Comment (JessterB) [inline data_analysis/agora/notebooks/preprocessing/preprocessing_utils.py]: nitpick - 'pathches' looks like a typo

File: data_analysis/agora/notebooks/preprocessing/preprocessing_utils.py
Comment (JessterB) [inline data_analysis/agora/notebooks/preprocessing/preprocessing_utils.py]: curiosity - who is manually creating chromosome names, and why are they doing that?

**Review by JessterB — APPROVED** [review]
This lgtm, although I did not exhaustively review the changes to the notebooks...nor am I at all sure that I'm competent to do so. I'm wondering if there's anyone else on our team that might be capable and willing to review your more scientific/specialized code...I know Laura is interested in developing her python skills, so maybe we could convince her that it would be a good learning experience!

File: data_analysis/agora/notebooks/preprocessing/preprocessing_utils.py
Comment (jaclynbeck-sage) [reply data_analysis/agora/notebooks/preprocessing/preprocessing_utils.py]: Yeah that's definitely a typo... oops. Will fix.

File: data_analysis/agora/notebooks/preprocessing/preprocessing_utils.py
Comment (jaclynbeck-sage) [reply data_analysis/agora/notebooks/preprocessing/preprocessing_utils.py]: Nobody is doing this. I don't know why, but I guess I added this to make the function sound 'generic'... Do you want me to just remove that part of the comment?

**Comment by jaclynbeck-sage** [comment]
I fixed the typo and also removed the comment about manually formatting chromosome names to avoid confusion. @JessterB do you want me to ask Laura to review the analysis part of the code before merging this PR?

**Comment by JessterB** [comment]
@jaclynbeck-sage Having Laura review the analysis would be great if she is willing and has the time. I don't want to block merging this for more than ~1 week though, since I'd like to get the updated SRM data ready for release.

**Comment by jaclynbeck-sage** [comment]
@JessterB I asked Laura and she said she's not at all familiar with Python. So I just described to her the steps I'm taking to process the data and she said it seemed sound. Not the same as a code review but, hopefully good enough for now?

**Comment by JessterB** [comment]
@jaclynbeck-sage Yes, that works. Thanks!

**Pull request merged**
